### ISSUE TITLE
SizeRollingFileTree exception

### ISSUE THREAD

**Issue by adrey** [magillus/flutter-fimber#86]
Hi! I'm trying to use `SizeRollingFileTree` in my app
```
void main() {
  Fimber.plantTree(DebugTree());
  Fimber.plantTree(SizeRollingFileTree(DataSize(megabytes: 1)) );
  Fimber.d("starting app");
```

pubspec.yaml:
```
dependencies:
  fimber_io: ^0.4.4
  fimber: ^0.4.4
```

and I've got this error on Android 10 device, storage permission granted
```
E/flutter (19744): [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: FileSystemException: Directory listing failed, path = 'log_/' (OS Error: No such file or directory, errno = 2)
E/flutter (19744): #0      _Directory._fillWithDirectoryListing (dart:io-patch/directory_patch.dart:37:68)
E/flutter (19744): #1      _Directory.listSync (dart:io/directory_impl.dart:243:5)
E/flutter (19744): #2      SizeRollingFileTree.detectFileIndex (package:fimber_io/src/file_log.dart:149:10)
E/flutter (19744): #3      new SizeRollingFileTree (package:fimber_io/src/file_log.dart:137:5)
E/flutter (19744): #4      main (package:myapp/main.dart:15:20)
E/flutter (19744): #5      _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:233:25)
E/flutter (19744): #6      _rootRun (dart:async/zone.dart:1190:13)
E/flutter (19744): #7      _CustomZone.run (dart:async/zone.dart:1093:19)
E/flutter (19744): #8      _runZoned (dart:async/zone.dart:1630:10)
E/flutter (19744): #9      runZonedGuarded (dart:async/zone.dart:1618:12)
E/flutter (19744): #10     _runMainZoned.<anonymous closure> (dart:ui/hooks.dart:225:5)
E/flutter (19744): #11     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:301:19)
E/flutter (19744): #12     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```
Shouldn't the directory be created by fimber?

**Comment by iulian0512** [comment]
same issue on android 9

**Comment by magillus** [comment]
I apologize for long wait.

The file rolling was created to be used in CLI/dart side, it worked at some point on Android but new permissions and storage changes in Android OS stopped working.

I will double check if that is something to be fixed, and make sure that the directory creation is working as expected on CLI/dart side.

**Comment by magillus** [comment]
Fixed in 0.6.3

However not sure if that will be working on Android due to those possible issues on storage permission.
Created #102 to do more tests in future.
